### PR TITLE
A tool to evaluate body sensation

### PR DESCRIPTION
**Pull request by JackFu123**

### CODE REVIEW

File: modules/tools/rosbag/body_sensation_evaluation.py [hunk 1]
```diff
@@ -0,0 +1,224 @@
+# -*- coding: utf-8 -*-
```
Comment (Capri2014) [inline modules/tools/rosbag/body_sensation_evaluation.py]: utf-8 ?

File: modules/tools/rosbag/body_sensation_evaluation.py [hunk 2]
```diff
@@ -0,0 +1,224 @@
...
+        """check driving mode according to timestamp"""
+        for mode in self.driving_mode:
+            if timestamp >= mode["start_time"] and timestamp <= mode["end_time"]:
+                if mode["status"] == Chassis.COMPLETE_AUTO_DRIVE:
+                    return True
+                else:
+                    return False
+        return False
+
+    def _bumps_rollback(self, bump_time):
```
Comment (Capri2014) [inline modules/tools/rosbag/body_sensation_evaluation.py]: use bumps_rollback_ instead

File: modules/tools/rosbag/body_sensation_evaluation.py [hunk 3]
```diff
@@ -0,0 +1,224 @@
...
+                            if self._check_status(t):
+                                self.auto_counts["turning"] += 1
+                            else:
+                                self.manual_counts["speed_up_2"] += 1
+                        self._turning_flag = 0
+                    elif abs(acc_x) >= -2:
+                        self._turning_flag = 1
+
+
+def main():
```
Comment (Capri2014) [inline modules/tools/rosbag/body_sensation_evaluation.py]: Add usage instruction here

File: modules/tools/rosbag/body_sensation_evaluation.py [hunk 4]
```diff
@@ -0,0 +1,224 @@
...
+                    else:
+                        self.manual_counts["bumps"] += 1
+                else:
+                    if self._speed_down_2_flag:
+                        if acc_y <= -4:
+                            self._speed_down_4_flag = 1
+                            continue
+                        if acc_y <= -2:
+                            continue
+                        if self._speed_down_4_flag == 1 and t - self._last_speed_down_4_time >=1:
```
Comment (Capri2014) [inline modules/tools/rosbag/body_sensation_evaluation.py]: Similar issue

File: modules/tools/rosbag/body_sensation_evaluation.py [hunk 5]
```diff
@@ -0,0 +1,224 @@
...
+                #时间戳
+                #dt = time.strftime('%Y-%m-%d %H:%M:%S',time.localtime(_t))
+                self.timestamp = t
+                if t - self._bump_lock_time <= 3:
+                    continue
+                acc_x = msg.pose.linear_acceleration.x
+                acc_y = msg.pose.linear_acceleration.y
+                acc_z = msg.pose.linear_acceleration.z
+                diff_bump_time = t - self._last_bump_time
+                if abs(acc_z) >= 2 and diff_bump_time >= 3:
```
Comment (Capri2014) [inline modules/tools/rosbag/body_sensation_evaluation.py]: make 2 and 3 const variable

File: modules/tools/rosbag/body_sensation_evaluation.py (hunk 2 above)
Comment (JackFu123) [reply modules/tools/rosbag/body_sensation_evaluation.py]: Could you explain why? This method should be protected.

File: modules/tools/rosbag/body_sensation_evaluation.py (hunk 3 above)
Comment (JackFu123) [reply modules/tools/rosbag/body_sensation_evaluation.py]: Usage instruction is in the front.

File: modules/tools/rosbag/body_sensation_evaluation.py (hunk 5 above)
Comment (JackFu123) [reply modules/tools/rosbag/body_sensation_evaluation.py]: python doesn't have "const" like cpp. Have to define a new class if const variable is needed.

File: modules/tools/rosbag/body_sensation_evaluation.py (hunk 1 above)
Comment (JackFu123) [reply modules/tools/rosbag/body_sensation_evaluation.py]: This line has been deleted

File: modules/tools/rosbag/body_sensation_evaluation.py (hunk 5 above)
Comment (Capri2014) [reply modules/tools/rosbag/body_sensation_evaluation.py]: similar to kMaxAccleration

File: modules/tools/rosbag/body_sensation_evaluation.py (hunk 2 above)
Comment (Capri2014) [reply modules/tools/rosbag/body_sensation_evaluation.py]: sorry my bad here

**Pull request merged**